lib/babe/babe.go

```diff
@@ -68,7 +68,92 @@ type ServiceConfig struct {
 	Telemetry          telemetry.Client
 }
 
-// NewService returns a new Babe Service using the provided VRF keys and runtime
+// ServiceIFace interface that defines methods available to BabeService
+type ServiceIFace interface {
+	Start() error
+	Stop() error
+	EpochLength() uint64
+	Pause() error
+	IsPaused() bool
+	Resume() error
+	SlotDuration() uint64
+}
+
+// ServiceBuilder interface to define the building of babe service
+type ServiceBuilder interface {
+	NewServiceIFace(cfg *ServiceConfig) (ServiceIFace, error)
+}
+
+var _ ServiceBuilder = (*Builder)(nil)
+
+// Builder struct to hold babe builder functions
+type Builder struct{}
+
+//NewServiceIFace returns a new Babe Service using the provided VRF keys and runtime
+func (Builder) NewServiceIFace(cfg *ServiceConfig) (ServiceIFace, error) {
+	if cfg.Keypair == nil && cfg.Authority {
+		return nil, ErrNoBABEAuthorityKeyProvided
+	}
+
+	if cfg.BlockState == nil {
+		return nil, ErrNilBlockState
+	}
```

Comment:
Should all these nil checks be on a method on `cfg`? Like `err = cfg.Validate()` would be nice

Reply:
Updated.